api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -685,4 +697,37 @@ private FirecloudWorkspaceId generateFirecloudWorkspaceId(String namespace, Stri
     resp.setItems(userRoles);
     return ResponseEntity.ok(resp);
   }
+
+  @Override
+  public ResponseEntity<WorkspaceResponseListResponse> getPublishedWorkspaces() {
+    WorkspaceResponseListResponse response = new WorkspaceResponseListResponse();
+    response.setItems(workspaceService.getPublishedWorkspaces());
+    return ResponseEntity.ok(response);
+  }
+
+  @Override
+  @AuthorityRequired({Authority.FEATURED_WORKSPACE_ADMIN})
+  public ResponseEntity<EmptyResponse> publishWorkspace(
+      String workspaceNamespace, String workspaceId) {
+    org.pmiops.workbench.db.model.Workspace dbWorkspace =
+        workspaceService.getRequired(workspaceNamespace, workspaceId);
+
+    dbWorkspace = workspaceService.setPublished(dbWorkspace, getRegisteredUserDomainEmail(), true);
+    dbWorkspace.setPublished(true);
```

Comment:
My previous comment was not addressed: why is this outside of the `setPublished` method?

Reply:
fixed it